### DISCUSSION TITLE
Integrating tower-layer

### DISCUSSION THREAD

**Discussion by milad-habibi** [apollographql/router#524] [General]
Hi 👋
I'm wondering would it be possible to share the timeline of merging the tower-layer branch into the main branch?
I personally am interested to contribute to this project if possible but not sure where to start. So wondering does the project has some kind of chat env like Slack or Discord so we can collaborate?

Thanks.

**Comment by abernix** [comment]
Hey @milad-habibi!

Good news! We merged the tower work into `main` a couple weeks ago with https://github.com/apollographql/router/commit/051f7bc63c5b3d7d254a38c7185f6db3f0aabc76 and and it's actually been the foundation for both the `0.1.0-alpha.6` and, now, `0.1.0-alpha.7` releases.

What sort of contributions were you thinking about?  Would you be able to start with a discussion about what you're thinking about here on GitHub?  This is ideally the (best) place for us to collaborate since it's where the team does most of their work and planning from, and it makes sure that the larger community can find the conversations and the context for those conversations to be successful in their own participation.

What'd you have in mind?

**Reply by milad-habibi** [reply]
Hey @abernix. Tnx for your reply. 
Cool yep just worked a little bit with it.

Well, I believe the lack of documentation is a major challenge for any new user so I can start by completing the docs.

Also, I tried to replace our gateway with the router and there are a couple of things. For example, it seems the input definition is not yet implemented in the Schema struct so I temporarily disabled the input validation to quickly fix the validation issue of queries that are using non-scalar inputs. Talking about `validate_value` function. I can create a PR for it.

Thank you all for your great job with this new router it rocks!
